### PR TITLE
refactor builtin 

### PR DESCRIPTION
**Pull request by siddontang**
move builtin function into builtin package

### CODE REVIEW

**Comment by ngaut** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

File: expression/builtin/builtin.go [hunk 1]
```diff
@@ -15,22 +15,39 @@
...
-	"github.com/juju/errors"
+const (
+	// ExprEvalFn is the key saving Call expression.
+	ExprEvalFn = "$fn"
+	// ExprEvalArgCtx is the key saving Context for a Call expression.
+	ExprEvalArgCtx = "$ctx"
+	// ExprAggDone is the key indicating that aggregate function is done.
+	ExprAggDone = "$aggDone"
+	// ExprEvalArgAggEmpty is the key to evaluate the aggregate function for empty table.
+	ExprEvalArgAggEmpty = "$agg0"
```
Comment (qiuyesuifeng) [inline expression/builtin/builtin.go]: `$agg0` is not good, rename another value?

Comment (siddontang) [inline expression/builtin/builtin.go]: It is the old code, change it in another PR if possible.

File: expression/builtin/groupby.go [hunk 2]
```diff
@@ -30,18 +30,21 @@ import (
 
 // see https://dev.mysql.com/doc/refman/5.7/en/group-by-functions.html
 
-type aggregateDistinct struct {
-	// now we have to use memkv Temp, later may be use map directly
+// AggregateDistinct handles distinct data for aggregate function: count, sum, avg, and group_concat.
+type AggregateDistinct struct {
+	// Distinct is a memory key-value map.
+	// Now we have to use memkv Temp, later may be use map directly
```
Comment (qiuyesuifeng) [inline expression/builtin/groupby.go]: Comment end of .
Line 31, use `See: xxxx`

File: expression/call.go [hunk 3]
```diff
@@ -123,14 +126,38 @@ func (c *Call) Eval(ctx context.Context, args map[interface{}]interface{}) (v in
 		a[i] = v
 	}
 
+	c.createDistinct()
```
Comment (qiuyesuifeng) [inline expression/call.go]: Why here`createDistinct` ?

**Comment by siddontang** [comment]
PTAL @qiuyesuifeng

**Pull request merged**